.github/workflows/ci.yml

```diff
@@ -91,10 +91,31 @@ jobs:
           node-version: ${{ env.PRIMARY_NODE_VERSION }}
       - name: Build
         uses: ./.github/actions/prepare-build
-
-      - name: Run Check
+      - name: Run Lint
         run: yarn ${{ matrix.lint-task }}
 
+  lint_with_build_from_root:
+    name: Lint From Root
+    # because we lint with our own tooling, we need to build
+    needs: [build]
+    runs-on: macos-latest
```

Comment:
> `macos-latest`

https://github.com/typescript-eslint/typescript-eslint/actions/runs/4825613286/jobs/8596705424?pr=6754

```plaintext
Run TYPESCRIPT_ESLINT_EXPERIMENTAL_TSSERVER=true npx eslint .

<--- Last few GCs --->

[233[8](https://github.com/typescript-eslint/typescript-eslint/actions/runs/4825613286/jobs/8596705424?pr=6754#step:5:9):0x5[9](https://github.com/typescript-eslint/typescript-eslint/actions/runs/4825613286/jobs/8596705424?pr=6754#step:5:10)76d80]    72536 ms: Mark-sweep 2020.0 (2088.4) -> 2011.9 (2080.8) MB, 1230.0 / 0.0 ms  (average mu = 0.618, current mu = 0.305) allocation failure; scavenge might not succeed
[2338:0x5976d80]    72619 ms: Scavenge 2021.0 (2081.3) -> 2018.5 (2081.3) MB, 17.1 / 0.0 ms  (average mu = 0.618, current mu = 0.305) allocation failure; 
[2338:0x5976d80]    73028 ms: Scavenge 2021.3 (2081.6) -> 2019.3 (2[10](https://github.com/typescript-eslint/typescript-eslint/actions/runs/4825613286/jobs/8596705424?pr=6754#step:5:11)2.6) MB, 372.5 / 0.0 ms  (average mu = 0.618, current mu = 0.305) allocation failure; 


<--- JS stacktrace --->

FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory
 1: 0xb7a940 node::Abort() [node]
```

https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources